src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IArgument.vb

```diff
@@ -835,5 +758,110 @@ BC30272: 'y' is not a parameter of 'Public Sub M2(x As Integer)'.
             VerifyOperationTreeAndDiagnosticsForTest(Of InvocationExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
         End Sub
 
+
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact>
+        Public Sub InOutConversion()
+            Dim source = <![CDATA[
+Class Program
+    Sub M1()
+        Dim x = 1.0
+        M2(x)'BIND:"M2(x)"
+    End Sub
+
+    Sub M2(ByRef Optional a As Integer = 0)
+    End Sub
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IInvocationExpression ( Sub Program.M2([ByRef a As System.Int32 = 0])) (OperationKind.InvocationExpression, Type: System.Void) (Syntax: 'M2(x)')
+  Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: Program) (Syntax: 'M2')
+  Arguments(1):
+      IArgument (ArgumentKind.Explicit, Matching Parameter: a) (OperationKind.Argument) (Syntax: 'x')
+        InConversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: True, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+        OutConversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: True, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+        ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Double) (Syntax: 'x')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of InvocationExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact>
+        Public Sub InOutConversionUserDefined()
```

Comment:
>InOutConversionUserDefined [](start = 19, length = 26)

It would be good to have tests for scenarios when standard conversions are involved into user-defined conversion, either for the input value, or for the method's return value. The cases when either BoundUserDefinedConversion.InConversionOpt, or BoundUserDefinedConversion.OutConversionOpt are not null.

Reply:
I have added a test in the last commit. Is that what you suggested?